Src/xWorks/UploadToWebonaryDlg.cs

```diff
@@ -300,46 +308,32 @@ private void helpButton_Click(object sender, EventArgs e)
 			ShowHelp.ShowHelpTopic(m_helpTopicProvider, "khtpUploadToWebonary");
 		}
 
-		/// <summary>
-		/// Add a message to the status area. Make sure the status area is redrawn so the
-		/// user can see what's going on even if we are working on something.
-		/// </summary>
-		public void UpdateStatus(string statusString)
+		private void closeButton_Click(object sender, EventArgs e)
 		{
-			outputLogTextbox.AppendText(Environment.NewLine + statusString);
-			outputLogTextbox.Refresh();
+			SaveToModel();
 		}
 
-		/// <summary>
-		/// Respond to a new status condition by changing the background color of the
-		/// output log.
-		/// </summary>
-		public void SetStatusCondition(WebonaryStatusCondition condition)
+		private void reportButton_Click(object sender, EventArgs e)
 		{
-			Color newColor;
-			switch (condition)
+			using(var dlg = new WebonaryLogViewer(Model.LastUploadReport))
 			{
-				case WebonaryStatusCondition.Success:
-					// Green
-					newColor = ColorTranslator.FromHtml("#b8ffaa");
-					break;
-				case WebonaryStatusCondition.Error:
-					// Red
-					newColor = ColorTranslator.FromHtml("#ffaaaa");
-					break;
-				case WebonaryStatusCondition.None:
-					// Grey
-					newColor = ColorTranslator.FromHtml("#dcdad5");
-					break;
-				default:
-					throw new ArgumentException("Unhandled WebonaryStatusCondition", nameof(condition));
+				dlg.ShowDialog();
 			}
-			outputLogTextbox.BackColor = newColor;
 		}
 
-		private void closeButton_Click(object sender, EventArgs e)
+		/// <summary>
+		/// Add a message to the status area. Make sure the status area is redrawn so the
+		/// user can see what's going on even if we are working on something.
+		/// </summary>
+		public void UpdateStatus(string statusString, WebonaryStatusCondition c)
 		{
-			SaveToModel();
+			// Set the status to the greater of the current or the new update
+			m_uploadStatus = (WebonaryStatusCondition)Enum.ToObject(typeof(WebonaryStatusCondition),
```

Comment:
why the call `Enum.ToObject` instead of just casting the result of Math.Max directly to the enum? I can't see any functional difference